### PR TITLE
Improve minikube tutorial

### LINKED ISSUE #1144: Use a single cilium DaemonSet file for standard k8s and minikube
- [x] test with recent minikube and check if etcd >=3.1.0
- [x] change the daemonset to use the available etcd address / configuration file.

### PR DESCRIPTION
**Pull request by aanm**
Fixes #1144

- Add a new DaemonSet and Config map so that users will only need to modify the config map.
- Removed kubeconfig dependency when running cilium in a pod (can still be overwrite if the user specifies the kubeconfig or kube-apiserver options)
- Rewrite minikube tutorial for the new daemonset
- Add service to RBAC rules

### CODE REVIEW

**Review by tgraf — DISMISSED** [review]

**Comment by aanm** [comment]
~~@tgraf this is still WIP, cilium doesn't connect to etcd for some reason, I'm still trying to figure that out :)~~

This PR is ready to be reviewed!

File: examples/kubernetes/cilium-config.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: cilium-config
+  namespace: kube-system
+data:
+  # If you do not want to run cilium in debug mode change it to false
+  debug: "true"
```
Comment (tgraf) [inline examples/kubernetes/cilium-config.yaml]: Let's move this down and default to `false`

File: examples/kubernetes/cilium-config.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  # This etcd-config contains the etcd endpoints of your cluster. If you use
+  # TLS please make sure you uncomment the ca-file
+  etcd-config: |-
+    ---
+    endpoints:
+    - http://127.0.0.1:2379
+    #
+    # In case you want to use TLS in etcd, uncomment the following line
+    # and add the certificate in cilium-etcd-secrets bellow
```
Comment (tgraf) [inline examples/kubernetes/cilium-config.yaml]: Can you add a comment when to use this over the secrets?

File: tests/k8s/cluster/cluster-manager.bash [hunk 3]
```diff
@@ -193,16 +196,50 @@ function install_kubeadm() {
 }
 
 function install_cilium_config(){
-    sudo mkdir -p /var/lib/cilium
+    cat > "${dir}/cilium-config.yaml" <<EOF
```
Comment (tgraf) [inline tests/k8s/cluster/cluster-manager.bash]: As discussed offline, let's use the template and then `sed` to adjust it so we test the yam template that we provide.

**Review by tgraf — CHANGES_REQUESTED** [review]

File: examples/kubernetes/cilium-config.yaml (hunk 2 above)
Comment (aanm) [reply examples/kubernetes/cilium-config.yaml]: It's not a matter of which one is better. If you uncomment this line bellow, then you need to add the secret on line 41. I'll clarify this a bit.

File: examples/kubernetes/cilium-config.yaml (hunk 1 above)
Comment (aanm) [reply examples/kubernetes/cilium-config.yaml]: Fixed

File: examples/kubernetes/cilium-config.yaml (hunk 2 above)
Comment (aanm) [reply examples/kubernetes/cilium-config.yaml]: Fixed

File: tests/k8s/cluster/cluster-manager.bash (hunk 3 above)
Comment (aanm) [reply tests/k8s/cluster/cluster-manager.bash]: Fixed

**Review by tgraf — APPROVED** [review]

**Comment by tgraf** [comment]
Awesome! 👍

**Comment by ianvernon** [comment]
This got merged while I was reviewing. Can you please address my comments in a follow-up branch?

**Comment by aanm** [comment]
@ianvernon which comments?

**Pull request merged**